scripts/pcmping

```diff
@@ -0,0 +1,166 @@
+        if have_received:
+            return pkt, exp_pkt
+    return None, None
+
+def get_portchannel(interface):
+    configdb = swsssdk.ConfigDBConnector()
+    configdb.connect()
+    portchannels = configdb.get_table("PORTCHANNEL")
+    for key, value in portchannels.iteritems():
+        if interface in value['members']:
+            return key, value
+    print ("Interface %s is not a portchannel member. Please Check!" % interface)
+    sys.exit(1)
+
+def get_portchannel_ipv4(portchannel_name):
+    configdb = swsssdk.ConfigDBConnector()
+    configdb.connect()
+    config = configdb.get_config()
+    portchannel_interfaces = config["PORTCHANNEL_INTERFACE"]
+    for key in portchannel_interfaces.keys():
+        pc, ip = key
+        ip = ip.split("/")[0]
+        if pc == portchannel_name and ipaddress.IPAddress(ip).version == 4:
+            return ip
+    print ("Portchannel %s doesn't have IPV4 address. Please Check!" % portchannel)
+    sys.exit(1)
+   
+def create_socket(interface, portchannel):
+    # create sockets
+    try:
+        sockets = []
+        for iface in portchannel['members']:
+            s = socket.socket(socket.AF_PACKET, socket.SOCK_RAW, socket.htons(ETH_P_IP))
+            s.bind((iface, 0))
+            s.settimeout(RCV_TIMEOUT)
+            if iface == interface:
+                exp_socket = s
+            sockets.append(s)
+    except:
+        print "Unable to create socket. Check your permissions"
+        sys.exit(1)
+    return sockets, exp_socket
+
+def main():
+    parser  = argparse.ArgumentParser(description='Check portchannel member link state',
+                                      version='1.0.0',
+                                      formatter_class=argparse.RawTextHelpFormatter,
+                                      epilog=""" 
+    Example:
+    pcmping -p Ethernet8 -ip 10.10.10.10
+    pcmping -p Ethernet8 -ip 10.10.10.10 -c 5
+    pcmping -p Ethernet8 -ip 10.10.10.10 -c 5 -t 20
+"""
+    )
+
+    parser.add_argument('-p', '--port', type=str, required=True, help='Portchannel member interface')
+    parser.add_argument('-ip','--decapip', type=str, required=True, help='Neighbor decap ip (usually same as neighbor loopback ip)') 
+    parser.add_argument('-c', '--count', type=int, help='Number of decap packets to be sent.', default=0)
+    parser.add_argument('-t', '--trial', type=int, help='Maximum trial times to create decap packet.\nPlease increase the value as the number of portchannel members increases', default=20)
+    args = parser.parse_args()
+    
+    interface =  args.port
+    decap_ip = args.decapip
+    exp_count = args.count
+    max_trial =  args.trial
+
+    portchannel_name, portchannel = get_portchannel(interface)
+    portchannel_ip = get_portchannel_ipv4(portchannel_name)
+    sockets, exp_socket = create_socket(interface, portchannel)
+    pkt, exp_pkt = create_decap_packet(interface, decap_ip, portchannel_ip, sockets, exp_socket, max_trial)
```

Comment:
Didn't quite get the logic of max_trial. Can you briefly explain?

Reply:
We randomly vary the src_ip of the decap packet to find out the src_ip/dst_ip combination that make packet come back via the same interface (due to ecmp). Max_trial is the maximum trial of varying src_ip.

Reply:
ok